From 52fe46e0e8d540336a4dcecde8d9e9c6cc234887 Mon Sep 17 00:00:00 2001
From: Even Rouault <even.rouault@spatialys.com>
Date: Sun, 16 Aug 2026 23:31:00 +0200
Subject: [PATCH] TIFFStartTile(): remove redundant tilewidth != 0 check, and
 add missing tilelength == 0 one

(cherry picked from commit 52fe46e0e8d540336a4dcecde8d9e9c6cc234887)
---
 libtiff/tif_read.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/libtiff/tif_read.c b/libtiff/tif_read.c
index 6c2e8ef..e432ded 100644
--- a/libtiff/tif_read.c
+++ b/libtiff/tif_read.c
@@ -1614,7 +1614,12 @@ static int TIFFStartTile(TIFF *tif, uint32_t tile)
         return 0;
     }
     tilemod = tile % howmany32;
-    if (tilemod == 0 || td->td_tilelength == 0)
+    if (td->td_tilelength == 0)
+    {
+        TIFFErrorExtR(tif, module, "Zero tilelength");
+        return 0;
+    }
+    if (tilemod == 0)
     {
         tif->tif_dir.td_row = 0;
     }
@@ -1632,7 +1637,7 @@ static int TIFFStartTile(TIFF *tif, uint32_t tile)
         return 0;
     }
     tilemod = tile % howmany32;
-    if (tilemod == 0 || td->td_tilewidth == 0)
+    if (tilemod == 0)
     {
         tif->tif_dir.td_col = 0;
     }
-- 
2.53.0

